src/ansys/pyensight/core/utils/omniverse_glb_server.py

```diff
@@ -399,55 +401,55 @@ def upload_file(self, glb_filename: str, timeline: List[float] = [0.0, 0.0]) ->
-                        duplicate = key
-                        break
-                var_pb.id = saved_id
-                var_pb.name = saved_name
-                # texture is never referenced, skip it
-                if duplicate is not None:
-                    self._glb_textures[material_index] = self._glb_textures[duplicate]
-                elif material_index != -1:
-                    # if not, record a new variable
-                    self._handle_update_command(cmd)
-                    self._glb_textures[material_index] = d
-
-            # GLB Scene -> DSG View
-            cmd, view_pb = self._create_pb("VIEW", parent_id=self._scene_id)
-            view_pb.lookat.extend([0.0, 0.0, -1.0])
-            view_pb.lookfrom.extend([0.0, 0.0, 0.0])
-            view_pb.upvector.extend([0.0, 1.0, 0.0])
-            view_pb.timeline.extend(timeline)
-            camera = self._gltf.cameras[0]
-            if camera.type == "orthographic":
-                view_pb.nearfar.extend(
-                    [float(camera.orthographic.znear), float(camera.orthographic.zfar)]
-                )
-            else:
-                view_pb.nearfar.extend(
-                    [float(camera.perspective.znear), float(camera.perspective.zfar)]
-                )
-                view_pb.fieldofview = camera.perspective.yfov
-                view_pb.aspectratio = camera.aspectratio.aspectRatio
-
-            self._handle_update_command(cmd)
+                    # does this Variable/texture already exist?
+                    duplicate = None
+                    saved_id = var_pb.id
+                    saved_name = var_pb.name
+                    for key, value in self._glb_textures.items():
+                        var_pb.name = value["pb"].name
+                        var_pb.id = value["pb"].id
+                        if value["pb"] == var_pb:
+                            duplicate = key
+                            break
+                    var_pb.id = saved_id
+                    var_pb.name = saved_name
+                    # if a new texture, add the Variable and create an index to the material
+                    if duplicate is None:
+                        self._handle_update_command(cmd)
+                        self._glb_textures[material_index] = d
+                    else:
+                        # create an additional reference to this variable from this material
+                        self._glb_textures[material_index] = self._glb_textures[duplicate]
 
-            # for present, just the default scene
             # GLB file: general layout
             # scene: "default_index"
             # scenes: [scene_index].nodes -> [node ids]
-            for node_id in self._gltf.scenes[self._gltf.scene].nodes:
-                self._walk_node(node_id, view_pb.id)
+            # was scene_id = self._gltf.scene
+            num_scenes = len(self._gltf.scenes)
+            for scene_idx in range(num_scenes):

```

Comment:
Walk all scenes.  One UPDATE_VIEW per scene.  Dynamically generate a timeline for the scene.